Plugins/NativeEngine/Source/NativeEngine.h

```diff
@@ -73,10 +73,13 @@ namespace Babylon
         };
 
         std::unordered_map<uint16_t, UniformValue> Uniforms{};
+        std::unordered_map<uint16_t, size_t> UniformMaxElementLength{};
 
         void SetUniform(bgfx::UniformHandle handle, gsl::span<const float> data, size_t elementLength = 1)
         {
             UniformValue& value = Uniforms[handle.idx];
+            size_t& maxLenght = UniformMaxElementLength[handle.idx];
+            elementLength = std::max(static_cast<size_t>(1), std::min(maxLenght, elementLength));
```

Comment:
What is the `std::max` for?

Reply:
In the case where the unordered_map was not yet initialized for the handle.idx value it will return 0 for maxLenght. std::max prevents the length value from been set to 0.

Reply:
Can that happen? Isn't this a problem if it happens?

Reply:
With the updated codebase this is no longer required. Now we check if UniformInfo exists in the unordered map, if it does it should have valid data.